### ISSUE TITLE
@Unique and nativeFindUnique do not agree on case-sensitivity

### ISSUE THREAD

**Issue by neroux** [objectbox/objectbox-java#935]
The `Unique` annotation appears to be case-sensitive whereas `findUnique` (respectively rather its native counterpart `nativeFindUnique`) is case-insensitive. The concrete issue is a class with a unique string field

    @Unique
    String field

I now seem to be able to store two instances with "Test" and "test" as values for `field` each and `put` does not complain. That would suggest `Unique` is case-sensitive. However when subsequently querying the database with `findUnique` it will throw an exception because it found both entries.

**Comment by greenrobot-team** [comment]
Please thumbs up https://github.com/objectbox/objectbox-java/issues/806.

**Comment by neroux** [comment]
Thanks for the link, but while this is related it does not appear to be the same issue.

I was not so much asking for `@Unique` to be case-insensitive but rather that `@Unique` and `put` follow the same logic. That's not so much a feature request but rather a bug IMHO as they simply do not play along.

**Comment by greenrobot-team** [comment]
Hm, maybe some explanation: `Query.findUnique()` simply checks if there is only a single result. Despite the same name, it is not intended to work together with the `@Unique` annotation.

**Comment by neroux** [comment]
But the question remains, why does `findUnique` consider "test" and "TEST" to be the same record and throws an exception when the same two records are perfectly acceptable for `@Unique`? It would seem the way cases are handled is not consistent across ObjectBox.

**Comment by greenrobot-team** [comment]
As I said `Query.findUnique()` checks if there is a single result to the query. So I guess change your condition on `field` to be case-sensitive (the default is to ignore case)?

https://github.com/objectbox/objectbox-java/blob/7e4e2e71ab5ce5c8bc54672c9e639a539a588073/objectbox-java/src/main/java/io/objectbox/query/QueryBuilder.java#L642-L652

**Comment by greenrobot-team** [comment]
There is also an example in the docs on how to use Strings as IDs: https://docs.objectbox.io/entity-annotations#object-ids-id

**Comment by neroux** [comment]
> As I said Query.findUnique() checks if there is a single result to the query.

That I got, but that's the issue, why does `findUnique` consider "test" and "TEST" as duplicates if they are completely separate entries for `@Unique` and `put`. That's what I meant by inconsistency.

> So I guess change your condition on field to be case-sensitive (the default is to ignore case)?

I have changed the logic since then, but I am pretty sure I had the default settings. So essentially specifying `StringOrder.CASE_SENSITIVE` should fix it in this case, right?

Thanks.
